pyro/primitives.py

```diff
@@ -124,6 +124,19 @@ def factor(name, log_factor):
     sample(name, unit_dist, obs=unit_value)
 
 
+def deterministic(name, value, event_dim=None):
+    """
+    Deterministic statement to add a site with `value` to the trace. This does not affect
+    the model density.
```

Comment:
Could you expand the docstring with information about what exactly is added to the trace and when someone might want to use `deterministic`?

Reply:
In case you find it useful, we needed this for the linear regression example to record the mean response, which will be a deterministic function of the coefficients, data and the intercept.